### ISSUE TITLE
Good Solutions To Lacon

### ISSUE THREAD

**Issue by zhangzhenyu13** [willxxy/lacon#1]
I am one of the authors of the Lacon Paper.

I would thank willxxy for contribution the code repo which contains an excellent implementation for the lacon. 

It would be better to re-parameter the method as
`def lacon_loss(instance_embeddings, pred_labels, class_labels, class_label_embeddings, temperature=0.07, reg_lambda=1.0, normalized=True)`
and once the `normalized=False`: please firstly normalize the embeddings via `torch.nn.LayerNorm(dim=emb_dim)`.

Another important thing worth mentioning is that the multi-head version ICL can perform better. 
You can implement this via projection-in(nn.Linear) and then reshape output tensors as [batch, head, dim-new], and then split via head dim to compute each single head loss separately.

**Comment by willxxy** [comment]
Thank you for the detailed information. I will keep them noted.
